File: kattis/basicprogramming1/BasicProgramming1.py
def t1(a):
  print("7")

def t2(a):
  if a[0] > a[1]:
    print("Bigger")
  elif a[0] < a[1]:
    print("Smaller")
  else:
    print("Equal")

def t3(a):
  ordered = sorted(a[:3])
  print(ordered[1])

def t4(a):
  print(sum(a))

def t5(a):
  print(sum(x for x in a if x % 2 == 0))

def t6(a):
  letters = [chr(97 + x % 26) for x in a]
  print("".join(letters))

def t7(a):
  i = 0
  visited = [False]*(len(a))
  while True:
    visited[i] = True
    i = a[i]
    if i > len(a)-1:
      print("Out")
      break
    elif i == len(a)-1:
      print("Done")
      break
    elif visited[i]:
      print("Cyclic")
      break

def main():
  n, t = input().split()
  a = [int(x) for x in input().split()]

  switch = {
    "1": t1,
    "2": t2,
    "3": t3,
    "4": t4,
    "5": t5,
    "6": t6,
    "7": t7
  }

  switch.get(t)(a)

main()

File: kattis/basicprogramming1/BasicProgramming1.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class BasicProgramming1 {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    String[] nt = in.nextLine().split(" ");
    int n = Integer.parseInt(nt[0]);
    int t = Integer.parseInt(nt[1]);
    String[] strings = in.nextLine().split(" ");
    List<Integer> ints = new ArrayList<Integer>();
    String[] alphabet = "abcdefghijklmnopqrstuvwxyz".split("");

    for (String s : strings) {
      ints.add(Integer.parseInt(s));
    }

    switch (t) {
      case 1:
        System.out.println("7");
        break;
      case 2:
        if (ints.get(0) > ints.get(1)) {
          System.out.println("Bigger");
        } else if (ints.get(0) < ints.get(1)) {
          System.out.println("Smaller");
        } else {
          System.out.println("Equal");
        }
        break;
      case 3:
        List<Integer> ordered = ints.subList(0, 3);
        Collections.sort(ordered);
        System.out.println(ordered.get(1));
        break;
      case 4:
        int sum = ints.stream().reduce(0, Integer::sum);
        System.out.println(sum);
        break;
      case 5:
        List<Integer> evens = ints.stream().filter(i -> i % 2 == 0).collect(Collectors.toList());
        int sumEven = evens.stream().reduce(0, (x, y) -> x + y);
        System.out.println(sumEven);
        break;
      case 6:
        List<String> letters = ints.stream().map(i -> alphabet[i % 26]).collect(Collectors.toList());
        System.out.println(String.join("", letters));
        break;
      case 7:
        boolean[] visited = new boolean[n];
        int i = 0;
        while (true) {
          visited[i] = true;
          i = ints.get(i);
          if (i > n - 1) {
            System.out.println("Out");
            break;
          } else if (i == n - 1) {
            System.out.println("Done");
            break;
          } else if (visited[i]) {
            System.out.println("Cyclic");
            break;
          }
        }
        break;
      default:
        break;
    }
  }
}